sdks/java/io/solr/src/test/java/org/apache/beam/sdk/io/solr/SolrIOTest.java

```diff
@@ -0,0 +1,262 @@
+import org.apache.solr.cloud.SolrCloudTestCase;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.cloud.ZkStateReader;
+import org.apache.solr.util.TimeOut;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A test of {@link SolrIO} on an independent Solr instance.
+ */
+@ThreadLeakFilters(defaultFilters = true, filters = {
+    BeamThreadsFilter.class
+})
+public class SolrIOTest extends SolrCloudTestCase{
+  private static final Logger LOG = LoggerFactory.getLogger(SolrIOTest.class);
+
+  private static final String SOLR_COLLECTION = "beam";
+  private static final int NUM_SHARDS = 3;
+  private static final long NUM_DOCS = 400L;
+  private static final int NUM_SCIENTISTS = 10;
+  private static final long BATCH_SIZE = 200L;
+
+  private static AuthorizedCloudSolrClient solrClient;
+  private static SolrIO.ConnectionConfiguration connectionConfiguration;
+
+  @Rule
+  public TestPipeline pipeline = TestPipeline.create();
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    String securityJson = "{"
+        + "'authentication':{"
+        + "  'blockUnknown': true,"
+        + "  'class':'solr.BasicAuthPlugin',"
+        + "  'credentials':{'solr':'orwp2Ghgj39lmnrZOTm7Qtre1VqHFDfwAEzr0ApbN3Y="
+        + " Ju5osoAqOX8iafhWpPP01E5P+sg8tK8tHON7rCYZRRw='}}"
+        + "}";
+    configureCluster(3)
+        .addConfig("conf", getFile("cloud-minimal/conf").toPath())
+        .configure();
+    ZkStateReader zkStateReader = cluster.getSolrClient().getZkStateReader();
+    zkStateReader.getZkClient()
+        .setData("/security.json", securityJson.getBytes(Charset.defaultCharset()), true);
+    Thread.sleep(1000);
+    TimeOut timeOut = new TimeOut(60, TimeUnit.SECONDS);
+    while (!timeOut.hasTimedOut()) {
+      if (zkStateReader.getClusterState().getLiveNodes().size() == 3) {
+        break;
+      } else {
+        Thread.sleep(100);
+      }
+    }
+    if (timeOut.hasTimedOut()) {
+      fail("Timeout waiting for nodes come back");
+    }
+    String zkAddress = cluster.getZkServer().getZkAddress();
+    connectionConfiguration = SolrIO.ConnectionConfiguration.create(zkAddress, SOLR_COLLECTION)
+        .withBasicCredentials("solr", "SolrRocks");
+    solrClient = connectionConfiguration.createClient();
+    SolrIOTestUtils.createCollection(SOLR_COLLECTION, NUM_SHARDS, 1, solrClient);
+    SolrIOTestUtils.insertTestDocuments(SOLR_COLLECTION, NUM_DOCS, solrClient);
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    solrClient.close();
+  }
+
+  @Before
+  public void before() throws Exception {
+    SolrIOTestUtils.clearCollection(SOLR_COLLECTION, solrClient);
```

Comment:
Hm weird: we create the collection and insert documents in `BeforeClass`, but clear the collection before every test? What's the point in inserting the documents then?